tardis/tardis_portal/api.py

```diff
@@ -415,7 +415,14 @@ def update_detail(self, object_list, bundle):  # noqa # too complex
         elif isinstance(bundle.obj, DatasetParameter):
             return False
         elif isinstance(bundle.obj, DataFile):
-            return False
+            dataset = DatasetResource.get_via_uri(DatasetResource(),
+                                                  bundle.data['dataset'],
+                                                  bundle.request)
+            return all([
+                bundle.request.user.has_perm('tardis_portal.change_dataset'),
```

Comment:
datafile permissions automatically "bubble up", a single check should be enough, so just:
`bundle.request.user.has_perm('tardis_portal.change_datafile')`